packages/react-utils/src/compose/renderShorthandChildren.tsx

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import { ObjectShorthandProps, ShorthandRenderFunction } from './types';
+
+/**
+ * A helper function to resolve the correct kind of shorthand children to render
+ */
+export const renderShorthandChildren = <TProps,>(
+  children: ObjectShorthandProps<TProps>['children'],
+  props: TProps,
+  Component: React.ElementType<TProps>,
+) => {
+  if (!children) {
+    return null;
+  }
+
+  if (typeof children === 'function') {
```

Comment:
**suggestion:** can we add proper guard for specifics scope? in this case guard that will approperly narrow union to correct type without need of having explicit assertions on multiple places?

```
// $ExpectType TProps['children'] | ShorthandRenderFunction<TProps>
children....

if(isChildrenShorhandFunction(children){
   // $ExpectType -> children === ShorthandRenderFunction<TProps>
   return children(Component, props)
}
```

Reply:
This file has been removed